<launch>
    <arg name="uav_id" default="1"/>
    <!-- 启动全局规划算法 -->
    <arg name="lidar_scan_topic_name" default="/prometheus/sensors/2Dlidar_scan"/>
    <node pkg="prometheus_global_planner" name="global_planner_main" type="global_planner_main" output="screen" launch-prefix="gnome-terminal --">
        <param name="uav_id" value="$(arg uav_id)"/>
        <!-- 2维高度，建议与起飞高度一致 -->
        <param name="global_planner/fly_height" value="1.0"/>
        <!-- 停止距离 -->
        <param name="global_planner/safe_distance" value="0.01"/>
        <param name="global_planner/time_per_path" value="1.0"/>
        <param name="global_planner/replan_time" value="2.0"/>
        <!-- 选择地图更新方式：　0代表全局点云，1代表局部点云，2代表激光雷达scan数据 -->
        <param name="global_planner/map_input_source" value="2"/>
        <remap from="/uav$(arg uav_id)/prometheus/global_planning/laser_scan" to="$(arg lidar_scan_topic_name)" />
        <param name="global_planner/sim_mode" value="true"/>
        <param name="global_planner/is_2D" value="true"/>
        <!-- 启发式函数系数 -->
        <param name="astar/lambda_heu" value="2.0"/>
        <param name="astar/lambda_cost" value="300.0"/>
        <!-- 最大搜索步数 -->
        <param name="astar/allocate_num" value="10000"/>
        <!-- 分辨率 -->
        <param name="map/resolution" value="0.1" type="double"/>
        <!-- 障碍物膨胀距离 -->
        <!-- 为飞机的轴距即可 -->
        <param name="map/inflate" value="0.4" type="double"/>
        <!-- 地图范围 -->
        <param name="map/origin_x" value="-15.0" type="double"/>
        <param name="map/origin_y" value="-15.0" type="double"/>
        <param name="map/origin_z" value="-0.5" type="double"/>
        <param name="map/map_size_x" value="30.0" type="double"/>
        <param name="map/map_size_y" value="30.0" type="double"/>
        <param name="map/map_size_z" value="3.0" type="double"/>

        <param name="map/border" value="true"/>
        <param name="map/inflate" value="0.3"/>
    </node>
    
	<!-- 启动rviz,设为false可关闭 -->
	<arg name="visualization" default="true"/>
	<group if="$(arg visualization)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find prometheus_gazebo)/config/rviz_config/rviz_config_astar.rviz" />
    </group>
</launch>

        <!-- 真实点云 -->
        <!-- <node pkg="prometheus_slam" type="pc2_publisher_node" name="pc2_publisher_node" output="screen">	
		    <param name="pcd_path" type="string" value="$(find prometheus_gazebo)/maps/obstacle.pcd" />
	    </node> -->